Sources/VLCMedia.m

```diff
@@ -269,13 +269,7 @@ - (instancetype)initWithURL:(NSURL *)anURL
         const char *url;
         VLCLibrary *library = [VLCLibrary sharedLibrary];
         NSAssert(library.instance, @"no library instance when creating media");
-
-        if (([[anURL absoluteString] hasPrefix:@"sftp://"]) ||
-            ([[anURL absoluteString] hasPrefix:@"smb://"])) {
-            url = [[[anURL absoluteString] stringByRemovingPercentEncoding] UTF8String];
-        } else {
-            url = [[anURL absoluteString] UTF8String];
-        }
+        url = [[[anURL absoluteString] stringByRemovingPercentEncoding] UTF8String];
```

Comment:
@tguillem When creating a NSURL with special characters (e.g. Chinese characters), you need to percent-encode the string. This removes any existing percent encoding if the user passes in one to prevent duplicate percent-encoding in the future.